Plugins/Public/conn_plugin/Main.cpp

```diff
@@ -177,25 +175,53 @@ void MoveClient(unsigned int client, unsigned int targetBase)
 	if (info.bBeamed)
 		return;
 
-	// No plugin handled it, do it ourselves.
-	unsigned int system;
+	uint system;
 	pub::Player::GetSystem(client, system);
-	Universe::IBase* base = Universe::get_base(targetBase);
-
-	pub::Player::ForceLand(client, targetBase); // beam
 
-	// if not in the same system, emulate F1 charload
-	if (base->iSystemID != system)
+	// No plugin handled it, do it ourselves.
+	Universe::IBase* base = Universe::get_base(targetBase);
+	if (base)
 	{
-		Server.BaseEnter(targetBase, client);
-		Server.BaseExit(targetBase, client);
-		wstring wscCharFileName;
-		HkGetCharFileName(ARG_CLIENTID(client), wscCharFileName);
-		wscCharFileName += L".fl";
-		CHARACTER_ID cID;
-		strcpy(cID.szCharFilename, wstos(wscCharFileName.substr(0, 14)).c_str());
-		Server.CharacterSelect(cID, client);
+		pub::Player::ForceLand(client, targetBase); // beam	// if not in the same system, emulate F1 charload
+		if (base->iSystemID != system)
+		{
+			Server.BaseEnter(targetBase, client);
+			Server.BaseExit(targetBase, client);
+			wstring wscCharFileName;
+			HkGetCharFileName(ARG_CLIENTID(client), wscCharFileName);
+			wscCharFileName += L".fl";
+			CHARACTER_ID cID;
+			strcpy(cID.szCharFilename, wstos(wscCharFileName.substr(0, 14)).c_str());
+			Server.CharacterSelect(cID, client);
+		}
 	}
+	else
+	{
+		//All else failed, move to proxy base
+		uint returnSys = HookExt::IniGetI(client, "conn.retsystembackup");
+		auto sysInfo = Universe::get_system(returnSys);
+		string scProxyBase = (string)((const char*)sysInfo->nickname) + "_proxy_base";
+		uint proxyBaseID;
+		if (pub::GetBaseID(proxyBaseID, scProxyBase.c_str()) == -4)
+		{
+			PrintUserCmdText(client, L"Return impossible, contact staff");
+			return;
+		}
+		PrintUserCmdText(client, L"Player base renamed/destroyed, ship redirected to a proxy base");
+		pub::Player::ForceLand(client, proxyBaseID); // beam
+		if (base->iSystemID != system)
+		{
+			Server.BaseEnter(targetBase, client);
```

Comment:
doesn't this stuff need to be proxyBaseID? but also, can't we move the ForceLand to after our `if (base) { ... } else { ... }` blocks and just do it once?